enhancements/api-review/custom-tags-aws.md

```diff
@@ -0,0 +1,125 @@
+
+### Goals
+
+ - the administrator or service (in the case of Managed OpenShift) installing OpenShift can pass an arbitrary
+   list of user-defined tags to the OpenShift Installer, and everything created by the installer and all other
+   bootstrapped components will apply those tags to all resources created in AWS, for the life of the cluster.
+ - tags must be applied at creation time, in an atomic operation. It isn't acceptable to create an object and,
+   after some period of time, apply the tags post-creation.
+
+### Non-Goals
+
+ - to reduce initial scope, tags are applied only at creation time and not reconciled. If an administrator manually
+   changes the tags stored in the infrastructure resource, behavior is undefined. See below.
+ - to reduce initial scope, we are not implementing this for clouds other than AWS. We will not take any actions
+   to prohibit that later.
+
+## Proposal
+
+New `experimentalPropagateUserTags` field added to `.platform.aws` of install config to indicate that the user tags should be applied to AWS
+resources created by in-cluster operators.
+
+If `experimentalPropagateUserTags` is true, install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io`.
+
+Add a new field `userTags` to `.status.aws` of the `infrastructure.config.openshift.io` type. Tags included in the
+`userTags` field will be applied to new resources created for the cluster. The `userTags` field will be populated by the installer only if the `experimentalPropagateUserTags` field is true.
+
+Installer will apply these tags to all AWS resources it creates with terraform (e.g. bootstrap and master EC2 instances) (already exists)
+
+All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api)
+will apply these tags to all AWS resources they create.
+
+userTags that are specified in the infrastructure resource will merge with userTags specified in an individual component. In the case where a userTag is specified in the infrastructure resource and there is a tag with the same key specified in an individual component, the value from the individual component will have precedence and be used.
+
+### User Stories
+
+https://issues.redhat.com/browse/SDE-1146 - IAM users and roles can only operate on resources with specific tags
+As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
+AWS resource tags.
+
+### Risks and Mitigations
+
+The userTags field is intended to be set at install time and immutable. If the userTags field is changed post-install, there is no guarantee about how an in-cluster operator will respond to the change. Some operators may reconcile the change and change tags on the AWS resource. Some operators may ignore the change. If tags are removed from userTags, the tag may or may not be removed from the AWS resource.
```

Comment:
I would like to see some of the expected scenarios documented during development/QA for each component that responds to these tags.